anillo/middlewares/cookies.py

```diff
@@ -0,0 +1,31 @@
+import uuid
+from werkzeug.http import dump_cookie
+from werkzeug._compat import to_unicode, text_type, try_coerce_native
+from werkzeug._internal import _cookie_parse_impl
+
+def _parse_cookie(header, charset='utf-8', errors='replace'):
+    # If the value is an unicode string it's mangled through latin1.  This
+    # is done because on PEP 3333 on Python 3 all headers are assumed latin1
+    # which however is incorrect for cookies, which are sent in page encoding.
+    # As a result we
+    if isinstance(header, text_type):
+        header = header.encode('latin1', 'replace')
+
+    def _parse_pairs():
+        for key, val in _cookie_parse_impl(header):
+            key = to_unicode(key, charset, errors, allow_none_charset=True)
+            val = to_unicode(val, charset, errors, allow_none_charset=True)
+            yield try_coerce_native(key), {"value": val}
+
+    return dict(_parse_pairs())
+
+
+def cookies_middleware(func):
+    def wrapper(request):
+        request.cookies = _parse_cookie(request.headers['Cookie'])
+        response = func(request)
+        if hasattr(response, 'cookies'):
+            for key, value in response.cookies.items():
+                response.headers['Set-Cookie'] = dump_cookie(key, **value)
```

Comment:
If a response has more that one cookie entry this looks like that it always overwrites the "set-cookie" header...